client/components/sites-popover/index.jsx

```diff
@@ -8,7 +8,6 @@ var React = require( 'react' ),
  * Internal dependencies
  */
 var Popover = require( 'components/popover' ),
-	sitesList = require( 'lib/sites-list' )(),
```

Comment:
Yes: `<SitesPopover>` needs to either use a sites list coming from a parent who is going to `observe` it and pass changes down to us, or `observe` it ourselves.  I don't have a preference on which way that is accomplished, this way just seemed simpler at the time.